### PR TITLE
[search] Added m_needAddress and m_needHighlighting to SearchParams.

### PR DESCRIPTION
**Pull request by ygorshenin**

### CODE REVIEW

File: search/search_params.hpp [hunk 1]
```diff
@@ -37,16 +35,26 @@ struct SearchParams
...
+
   // A minimum distance between search results in meters, needed for
   // pre-ranking of viewport search results.
   double m_minDistanceOnMapBetweenResults = 0.0;
 
   Mode m_mode = Mode::Everywhere;
   bool m_forceSearch = false;
-  bool m_suggestsEnabled = true;
+
+  // Is it needed to generate search suggests.
```
Comment (mpimenov) [inline search/search_params.hpp]: s/Is it needed/Needed/

Comment (mpimenov) [inline search/search_params.hpp]: Please write also what m_forceSearch is needed for.

File: map/framework.cpp [hunk 2]
```diff
@@ -1343,8 +1344,11 @@ bool Framework::SearchEverywhere(search::EverywhereSearchParams const & params)
   p.m_inputLocale = params.m_inputLocale;
   p.m_mode = search::Mode::Everywhere;
   SetViewportIfPossible(p);  // Search request will be delayed if viewport is not available.
+  p.m_maxNumResults = search::SearchParams::kDefaultNumResultsEverywhere;
   p.m_forceSearch = true;
   p.m_suggestsEnabled = true;
+  p.m_needAddress = true;
+  p.m_needHighlight = true;
```
Comment (mpimenov) [inline map/framework.cpp]: nit: m_needHighlighting

At first I used m_needHighlights but "ing" seemed better.

Comment (ygorshenin) [reply map/framework.cpp]: Done

File: search/search_params.hpp (hunk 1 above)
Comment (ygorshenin) [reply search/search_params.hpp]: Done

Comment (ygorshenin) [reply search/search_params.hpp]: Done

File: search/processor.hpp [hunk 3]
```diff
@@ -140,16 +115,13 @@ class Processor : public my::Cancellable
...
-  double m_minDistanceOnMapBetweenResults;
-  Mode m_mode;
-  bool m_suggestsEnabled;
-  shared_ptr<hotels_filter::Rule> m_hotelsFilter;
-  bool m_cianMode = false;
-  SearchParams::OnResults m_onResults;
-
-  bool m_viewportSearch;
+  bool m_needAddress = true;
+  bool m_needHighlight = true;
```
Comment (mpimenov) [inline search/processor.hpp]: here too, please
and in the commit message

Comment (ygorshenin) [reply search/processor.hpp]: These fields must be removed.

**Pull request merged**